api/host_test.go

```diff
@@ -189,6 +189,22 @@ func TestWorkingStatus(t *testing.T) {
 		t.Fatal(err)
 	}
 
+	// Block until the allowance has finished forming contracts.
+	err = build.Retry(50, time.Millisecond*250, func() error {
+		var rc RenterContracts
+		err = st.getAPI("/renter/contracts", &rc)
+		if err != nil {
+			return errors.New("couldn't get renter stats")
+		}
+		if len(rc.Contracts) != 1 {
+			return errors.New("no contracts")
+		}
+		return nil
+	})
+	if err != nil {
+		t.Fatal("allowance setting failed")
+	}
```

Comment:
Had to add this boilerplate code to a lot of the tests to prevent NDFs. We should probably create that global test-helper package, there's sooo much copy-paste in our test suite, and a lot of the copy-pasted code isn't even correct.